### PR TITLE
Preserve data

### PR DESCRIPTION
**Pull request by Marlyn-Mayienga**
In this module, we:
- Modified the contact form so that it could store information on local storage
- The form can allow the user to load the page, if there is any data in the local storage the input fields are pre-filled with this data.

### CODE REVIEW

File: main.js [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+    errorMessage.textContent =
+      "Most write a message of less than 1000 characters";
+    errorMessage.classList.add("show");
+    e.preventDefault()
+    return;
+
+  }else{
+    errorMessage.classList.remove("show");
+  }
+  // document.getElementById("form").submit()
```
Comment (codecaiine) [inline main.js:88]: - [ ] I would like to recommend you remove all the commented code from your codebase in order to keep your code clean :+1:

File: main.js [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+    ) {
+      return true;
+    }
+  }
+  return false;
+}
+form.addEventListener("submit", handleSubmit);
+
+function handleSubmit(e){
+  // e.preventDefault()
```
Comment (codecaiine) [inline main.js:39]: - [ ]  I would like to recommend you remove all the commented code from your codebase in order to keep your code clean :smile:

File: main.js [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+  if (message.value.length > 1000 || message.value === "") {
+    errorMessage.textContent =
+      "Most write a message of less than 1000 characters";
+    errorMessage.classList.add("show");
+    e.preventDefault()
+    return;
+
+  }else{
+    errorMessage.classList.remove("show");
+  }
```
Comment (codecaiine) [inline main.js:87]: - [ ] Great setup of your local Storage :clap: But there are some issues you need to address to make it great :disappointed: Your contact form data is not stored in your local storage, Please create one object to store all the fields data from the requirement every input field need to be store in a single object and please ensure that After filling the form, and not submitting it, the entered values are saved in the browser.:smile: 

- [ ] Please make sure that when the data are retrieved from the localStorage is not vanishing after reloading the page. That means the input fields are pre-filled with the data  even when you are reloading the page :+1:

**Review by codecaiine — CHANGES_REQUESTED** [review]
Hi @Marlyn-Mayienga @LokurasRlz , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
- Descriptive pull request :+1: 
- No linters errors :heavy_check_mark: 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me (@codecaiine) in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: main.js [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+
+form2.addEventListener('submit', (e) => {
+  e.preventDefault();
+  const obj = {
+    fullname: document.querySelector('#username').value,
+    EmailAdress: document.querySelector('#email').value,
+    message: document.querySelector('#msg').value,
+  };
+
+  localStorage.setItem('DATA', JSON.stringify(obj));
```
Comment (HAMMAS-SALEEM) [inline main.js:103]: - [ ] It is great that you are working with localStorage but your form should be submitted after setting data to local storage :+1: You can use `form.submit()` method for submitting form after setting it to the localStorage :blush:

**Review by HAMMAS-SALEEM — CHANGES_REQUESTED** [review]
# Required Changes :recycle:
almost-there

### Hey @Marlyn-Mayienga, @LokurasRlz :wave:

### Highlights :1st_place_medal:
 
:heavy_check_mark: Following HTML, CSS best practices
:heavy_check_mark: Following Project Requirements
:heavy_check_mark: Descriptive PR title
:heavy_check_mark: Descriptive branch name

Good job on implementing some of the changes requested by the previous reviewer, there are some issues that you still need to work on to go to the next project but you are almost there! :blush:

# Optional suggestions
[Optional]
- [ ] Please add eslint linter in your linters.yml file because it is very important for your javascript files. It checks for the issues in the javascript files which can break the application :fearful:

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding :clap: :clap: :clap:

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @HAMMAS-SALEEM in your question so I can receive the notification.

_**Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise.**_

------------

_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by ishimwezachee — CHANGES_REQUESTED** [review]
Hi @Marlyn-Mayienga , 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


 ## Highlights
✔️ Following HTML, CSS best practices
✔️ Following Project Requirements
✔️ Descriptive PR title
✔️ Descriptive branch name

 ## Required Changes ♻️

- [ ]  You should implement the changes request by previous reviewer. As he suggested, you data should be stored on local storage after submission, but right now if I submit the data, in local storage, there is not data. its empty, please implement it by just [storing ](https://www.w3schools.com/jsref/prop_win_localstorage.asp)those data before submit form function is called. 

_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: main.js [hunk 5]
```diff
@@ -18,72 +17,68 @@ function toggleMenu() {
...
+
+window.addEventListener('load', () => {
+  if (localStorage.getItem('DATA')) {
+    document.querySelector('#username').value = getDataValue.fullname;
+    document.querySelector('#email').value = getDataValue.emailAddress;
+    document.querySelector('#msg').value = getDataValue.message;
   }
-  // document.getElementById("form").submit()
-}
+});
```
Comment (Kandy-Peter) [inline main.js:84]: - [ ] Actually, the changes you made are not related to the one requested by the previous reviewer, you have to make sure that input values are stored locally even if the form is not submitted. This means that you'll target each input value and save it to the localStorage the way that even if the user left the page and come back, the form will be filled with the previous data. Kindly, implement these changes. 

Hint: 
1. Store inputs values in an object. eg: 
  ```
     const formData = {
       name: nameInput.value,
       email: emailInput.value,
       body: bodyInput.value,
     };
  ```
2. Use `localStorage.setItem` and `localStorage,getItem` to set and get data.
3. On window load get the store data from the local storage.

**Review by Kandy-Peter — CHANGES_REQUESTED** [review]
Hi @Marlyn-Mayienga

Good job so far! I know you have worked hard on implementing the project and its requirements. However, there is an issue that you still need to work on to go to the next project but you are almost there!

## To highlight :champagne: 
- All linters are passing :heavy_check_mark: 
- You  are following GitHub actions :heavy_check_mark: 
- Successfully deployed :heavy_check_mark: 
## Required Changes ♻️

- [x] Kindly check the comments under the review.

## Optional suggestions

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by leonardodiasb — APPROVED** [review]
# Status: ✔️ Project Approved ✔️ 

Hello @Marlyn-Mayienga 🔥 

You worked hard in all the instructions that the previous Code Reviewers give you👍🏽 

Now you can merge this branch 🚀 

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**